### PR TITLE
Fix: Versioning spread in non-versioned namespace & many-one version mapping

### PR DESCRIPTION
**Pull request by timotheeguerin**

### CODE REVIEW

File: packages/versioning/src/versioning.ts [hunk 1]
```diff
@@ -352,32 +352,24 @@ interface VersionProjections {
...
-    return { version: resolution.rootVersion?.value, projections };
+    // const projections = [...resolution.versions.entries()].map(([ns, version]) => {
+    //   return;
+    // });
+    return {
+      version: resolution.rootVersion?.value,
+      projections: [
+        {
+          projectionName: "v",
+          arguments: [resolution.versions as any],
```
Comment (timotheeguerin) [inline packages/versioning/src/versioning.ts:369]: @bterlson a few things here:
1. we don't allow non primitive types in the TS type
2. if you bypass it produce a type: "Object" with properties being what you passed but there is no type for it

File: packages/versioning/src/versioning.ts [hunk 2]
```diff
@@ -352,32 +352,24 @@ interface VersionProjections {
...
   return resolutions.map((resolution) => {
-    const projections = [...resolution.versions.entries()].map(([ns, version]) => {
-      return {
-        scope: ns,
-        projectionName: "v",
-        arguments: [version.enumMember],
-      };
-    });
-    return { version: resolution.rootVersion?.value, projections };
+    // const projections = [...resolution.versions.entries()].map(([ns, version]) => {
```
Comment (timotheeguerin) [inline packages/versioning/src/versioning.ts:361]: remove

File: packages/versioning/src/versioning.ts [hunk 3]
```diff
@@ -484,16 +480,21 @@ function appliesAtVersion(
...
-    }
+  if (namespace === undefined) {
+    return null;
+  }
+  // let version = getVersionForEnumMember(p, enumMemberVersion)!;
+  const version = enumMemberVersion.properties.get(namespace.projectionBase ?? namespace);
+  if (version === undefined) {
+    return null;
   }
+  // const newVersion = versionIndex.get(version)?.get(namespace);
```
Comment (timotheeguerin) [inline packages/versioning/src/versioning.ts:494]: remove

File: packages/versioning/src/versioning.ts [hunk 4]
```diff
@@ -484,16 +480,21 @@ function appliesAtVersion(
   getMetadataFn: (p: Program, t: Type) => Version | undefined,
   p: Program,
   type: Type,
-  enumMemberVersion: EnumMemberType
-) {
+  enumMemberVersion: any
```
Comment (timotheeguerin) [inline packages/versioning/src/versioning.ts:483]: update with real types as well as `addedAfter`, and others

**Review by markcowl — APPROVED** [review]
Modulo the outstanding work, this looks good to me.

**Review by bterlson — APPROVED** [review]
Seems great, nice work!

**Pull request merged**